aten/src/ATen/native/native_functions.yaml

```diff
@@ -602,6 +602,38 @@
     CUDA: logical_xor_out
   supports_named_tensor: True
 
+- func: logical_and(Tensor self, Tensor other) -> Tensor
+  use_c10_dispatcher: unboxed_only
+  variants: function, method
+  supports_named_tensor: True
```

Comment:
since these support naming, we should have tests in test_namedtensor.py like we have for test_logical_xor.